### PR TITLE
Aatifsyed/2883 build profiles are too aggressive

### PR DESCRIPTION
**Pull request by aatifsyed**
## Summary of changes

Changes introduced in this pull request:

- Don't abort the process on panic
- Keep debug info in debug builds
- Remove unused `quick` profile

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -43,7 +43,6 @@ members = [
   "utils/forest_utils",
   "utils/forest_shim",
 ]
-resolver = "2"
```
Comment (aatifsyed) [inline Cargo.toml:46]: this is implied

File: Cargo.toml [hunk 2]
```diff
@@ -191,16 +190,9 @@ forest_test_utils = { path = "./utils/test_utils" }
...
 
 [profile.dev]
-debug = 0
 split-debuginfo = "unpacked"
 
-[profile.quick]
-inherits = "release"
-opt-level = 1
-lto = "off"
-
```
Comment (aatifsyed) [inline Cargo.toml:201]: does anyone actually use this?

File: Cargo.toml [hunk 3]
```diff
@@ -191,16 +190,9 @@ forest_test_utils = { path = "./utils/test_utils" }
 forest_utils = { path = "./utils/forest_utils" }
 
 [profile.dev]
-debug = 0
```
Comment (aatifsyed) [inline Cargo.toml:194]: why wouldn't we want debug info in a dev build?

File: Cargo.toml (hunk 1 above)
Comment (aatifsyed) [reply Cargo.toml:46]: Thanks!

File: Cargo.toml (hunk 3 above)
Comment (LesnyRumcajs) [reply Cargo.toml:194]: @aatifsyed https://github.com/ChainSafe/forest/pull/2918

Long story short, a `make test-all` would bloat to 71G (or more) with this. Given the overhead and how seldom one needs to go full GDB, it makes sense to disable it. Do you agree?

**Pull request closed without merging**